couchbase.go

```diff
@@ -43,81 +42,92 @@ type RolesAndGroups struct {
 func New() (interface{}, error) {
 	db := new()
 	// Wrap the plugin with middleware to sanitize errors
-	dbType := dbplugin.NewDatabaseErrorSanitizerMiddleware(db, db.secretValues)
+	dbType := newdbplugin.NewDatabaseErrorSanitizerMiddleware(db, db.secretValues)
 	return dbType, nil
 }
 
 func new() *CouchbaseDB {
 	connProducer := &couchbaseDBConnectionProducer{}
 	connProducer.Type = couchbaseTypeName
 
-	credsProducer := &credsutil.SQLCredentialsProducer{
-		DisplayNameLen: 50,
-		RoleNameLen:    50,
-		UsernameLen:    50,
-		Separator:      "-",
-	}
-
 	db := &CouchbaseDB{
 		couchbaseDBConnectionProducer: connProducer,
-		CredentialsProducer:           credsProducer,
 	}
 
 	return db
 }
 
 // Run instantiates a CouchbaseDB object, and runs the RPC server for the plugin
-func Run(apiTLSConfig *api.TLSConfig) error {
-	dbType, err := New()
+func Run() error {
+	db, err := New()
 	if err != nil {
 		return err
 	}
 
-	dbplugin.Serve(dbType.(dbplugin.Database), api.VaultPluginTLSProvider(apiTLSConfig))
+	newdbplugin.Serve(db.(newdbplugin.Database))
 
 	return nil
 }
 
-func (c *CouchbaseDB) Type() (string, error) {
-	return couchbaseTypeName, nil
+func (c *CouchbaseDB) Initialize(ctx context.Context, req newdbplugin.InitializeRequest) (newdbplugin.InitializeResponse, error) {
+	err := c.couchbaseDBConnectionProducer.Initialize(ctx, req.Config, req.VerifyConnection)
+	if err != nil {
+		return newdbplugin.InitializeResponse{}, err
+	}
+	resp := newdbplugin.InitializeResponse{
+		Config: req.Config,
+	}
+	return resp, nil
 }
 
-func computeTimeout(ctx context.Context) (timeout time.Duration) {
-	deadline, ok := ctx.Deadline()
-	if ok {
-		return time.Until(deadline)
+func (c *CouchbaseDB) NewUser(ctx context.Context, req newdbplugin.NewUserRequest) (newdbplugin.NewUserResponse, error) {
+	// Grab the lock
+	c.Lock()
+	defer c.Unlock()
+
+	statements := removeEmpty(req.Statements.Commands)
+	if len(statements) == 0 {
+		return newdbplugin.NewUserResponse{}, dbutil.ErrEmptyCreationStatement
```

Comment:
I was unsure what the desired outcome would be. I wasn't sure if we wanted to keep the behaviors intact or were changing them to a new standard. As such, I can restore these behaviors, as well as restoring the test.

Reply:
I lean towards having it be a required field for all databases as a default set of statements usually doesn't make sense since there's usually some amount of permissions to be set on the user.

Reply:
This was reverted to the original behavior in my most recent commits.